### PR TITLE
IGNITE-12262 Implement a Pub/Sub Data Streamer 

### PR DESCRIPTION
**Pull request by gkatzioura**
Based on the issue https://issues.apache.org/jira/browse/IGNITE-12262.
This is a fork from the Flink branch.

### CODE REVIEW

File: modules/pub-sub/pom.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    <artifactId>ignite-pub-sub</artifactId>
+    <version>2.9.0-SNAPSHOT</version>
+    <url>http://ignite.apache.org</url>
+
+    <properties>
+        <pubsub.version>1.60.0</pubsub.version>
+        <ignite-core.version>2.7.6</ignite-core.version>
+        <ignite-log4j.version>2.7.6</ignite-log4j.version>
+        <ignite-spring.version>2.7.6</ignite-spring.version>
```
Comment (samaitra) [inline modules/pub-sub/pom.xml]: please update the dependency version for Ignite libs to `2.9.0-SNAPSHOT`

File: modules/pub-sub/pom.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        <version>1</version>
+        <relativePath>../../parent</relativePath>
+    </parent>
+
+    <artifactId>ignite-pub-sub</artifactId>
+    <version>2.9.0-SNAPSHOT</version>
+    <url>http://ignite.apache.org</url>
+
+    <properties>
+        <pubsub.version>1.60.0</pubsub.version>
```
Comment (samaitra) [inline modules/pub-sub/pom.xml]: can we update the pubsub lib dependency to `1.102.0`

File: modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/MockPubSubServer.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (samaitra) [inline modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/MockPubSubServer.java]: Do we need the copyright info or Apache 2.0 licence will suffice?

File: modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/PubSubStreamerSelfTest.java [hunk 4]
```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (samaitra) [inline modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/PubSubStreamerSelfTest.java]: Do we need the copyright info or Apache 2.0 licence will suffice?

File: modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/PubSubStreamerSelfTest.java [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+     * @throws TimeoutException If timed out.
+     * @throws InterruptedException If interrupted.
+     */
+    private void consumerStream(ProjectTopicName topic, Map<String, String> keyValMap)
+            throws InterruptedException, IOException {
+        PubSubStreamer<String, String> pubSubStmr = null;
+
+        try (IgniteDataStreamer<String, String> stmr = ignite.dataStreamer(DEFAULT_CACHE_NAME)) {
+            stmr.allowOverwrite(true);
+            stmr.autoFlushFrequency(10);
```
Comment (samaitra) [inline modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/PubSubStreamerSelfTest.java]: can we use the constant `MESSAGES_PER_REQUEST ` here as well?

File: modules/pub-sub/src/main/java/org/apache/ignite/stream/pubsub/PubSubStreamer.java [hunk 6]
```diff
@@ -0,0 +1,231 @@
+package org.apache.ignite.stream.pubsub;
+
```
Comment (samaitra) [inline modules/pub-sub/src/main/java/org/apache/ignite/stream/pubsub/PubSubStreamer.java]: Please add apache license

File: modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/PubSubStreamerTestSuite.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package org.apache.ignite.stream.pubsub;
```
Comment (samaitra) [inline modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/PubSubStreamerTestSuite.java]: Please add apache licence

File: modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/PubSubStreamerSelfTest.java (hunk 4 above)
Comment (gkatzioura) [reply modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/PubSubStreamerSelfTest.java]: Removed it and set the Apache 2.0 licence

File: modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/MockPubSubServer.java (hunk 3 above)
Comment (gkatzioura) [reply modules/pub-sub/src/test/java/org/apache/ignite/stream/pubsub/MockPubSubServer.java]: Removed it and set the Apache 2.0 licence

**Pull request closed without merging**